### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by Runi-VN**
I spent a lot of time on:
- Trying to configure the console logger
- Trying to configure saving files in another place, e.g. through HTTP/FTP (Failed)

## When reviewing
Please let me know if you agree with where I inserted the loggers in `server.ts`.

## To do
- [ ] Write about the logger in the report
Why we didn't save it away from the app ;(
-  ~~Investigate the errorloggers level of verbosity~~

## Info
To understand the formats and fields, please see:  
https://github.com/winstonjs/logform#formats  
https://www.npmjs.com/package/express-winston

**I added the [debug](https://www.npmjs.com/package/debug) package**

#### Possible additions in the future:
https://github.com/winstonjs/winston/blob/master/docs/transports.md#mongodb-transport
https://github.com/winstonjs/winston/blob/master/docs/transports.md#http-transport
https://github.com/ducas/apollo-logging-winston
https://www.npmjs.com/package/express-winston-middleware

### CODE REVIEW

File: Backend/package.json [hunk 1]
```diff
@@ -22,22 +22,25 @@
...
+    "winston": "^3.2.1"
   },
   "devDependencies": {
     "@types/compression": "^1.7.0",
+    "@types/cors": "^2.8.6",
     "@types/express": "^4.17.6",
     "@types/graphql": "^14.5.0",
     "@types/graphql-depth-limit": "^1.1.2",
     "@types/node": "^14.0.1",
-    "@types/cors": "^2.8.6",
```
Comment (Runi-VN) [inline Backend/package.json:40]: idk why this was removed? merge?

**Comment by MalteMagnussen** [comment]
Where does it decide where to store the logs?

File: Backend/src/middlewares/logger.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import path from "path"
+
+let requestLoggerTransports: Array<any> = [
+    //always save to file
+    new winston.transports.File({ filename: path.join(process.cwd(), "logs", "requests.log") })
+]
+let errorLoggerTransports: Array<any> = [
+    //always save to file
+    new winston.transports.File({ filename: path.join(process.cwd(), "logs", "error.log") })
+]
```
Comment (MalteMagnussen) [inline Backend/src/middlewares/logger.ts:12]: So this is where it makes the file, where it stores the log, right? 
Where does this file go?

File: Backend/src/middlewares/logger.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    errorLoggerTransports.push(new winston.transports.Console({
+        format: format.combine(
+            //format.colorize({ all: true }),
+            format.json(),
+            //format.errors({ stack: true }), //print stacktrace to console
+            format.simple(),
+            format.prettyPrint()
+        )
+    }));
+}
```
Comment (MalteMagnussen) [inline Backend/src/middlewares/logger.ts:33]: This means that it is _only_ logging in Production environments?

File: Backend/src/middlewares/logger.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+})
+
+let errorLogger = expressWinston.errorLogger({
+    transports: errorLoggerTransports,
+    format: format.combine(
+        format.json(),
+        format.timestamp({ format: 'YYYY-MM-DD HH:mm:ss' })
+    ),
+
+})
```
Comment (MalteMagnussen) [inline Backend/src/middlewares/logger.ts:56]: So its saved in [JSON](https://github.com/winstonjs/logform#json) format + a timestamp?

**Review by MalteMagnussen — APPROVED** [review]

File: Backend/src/middlewares/logger.ts (hunk 2 above)
Comment (Runi-VN) [reply Backend/src/middlewares/logger.ts:12]: root/logs/filename

File: Backend/src/middlewares/logger.ts (hunk 3 above)
Comment (Runi-VN) [reply Backend/src/middlewares/logger.ts:33]: Only in development do we also log to console. When the application is deployed we only log to file, as is the default.

File: Backend/src/middlewares/logger.ts (hunk 4 above)
Comment (Runi-VN) [reply Backend/src/middlewares/logger.ts:56]: Yes, and the timestamp is in that specific format.

**Review by Castau — APPROVED** [review]

**Pull request merged**